script.service.ltvpl/util.py

```diff
@@ -0,0 +1,139 @@
+#
+#       Copyright (C) 2018
+#       John Moore (jmooremcc@hotmail.com)
+#
+#  This Program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2, or (at your option)
+#  any later version.
+#
+#  This Program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with XBMC; see the file COPYING.  If not, write to
+#  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+#  http://www.gnu.org/copyleft/gpl.html
+#
+
+import xbmc
+import xbmcaddon
+import xbmcgui
+import os
+import sys
+from datetime import datetime
+
+__Version__ = "1.1.2"
+
+PYVER = float('{}.{}'.format(*sys.version_info[:2]))
+
+def GetXBMCVersion():
+    version = xbmcaddon.Addon('xbmc.addon').getAddonInfo('version')
+    version = version.split('.')
+    return int(version[0]), int(version[1]) #major, minor
+
+
+
+ADDON   =  xbmcaddon.Addon()
+ADDONID = ADDON.getAddonInfo('id')
+ADDON_NAME = ADDON.getAddonInfo('name')
+if PYVER < 3.0:
+    ADDON_PATH = ADDON.getAddonInfo('path').decode('utf-8')
+else:
+    ADDON_PATH = ADDON.getAddonInfo('path')
+
+ADDON_USERDATA_FOLDER = xbmc.translatePath("special://profile/addon_data/"+ADDONID)
+BASEPATH = os.path.join(ADDON_PATH,r"resources")
+if BASEPATH not in sys.path:
+    sys.path.insert(2,BASEPATH)
```

Comment:
Please don't use `sys.path`, but give your add-on a proper structure so you can use things like `from resources.lib.data...... import .....`

Reply:
Removed the code setting sys.path